src/components/NewExperiment.vue

```diff
@@ -1717,7 +1787,32 @@ export default Vue.extend({
         dialogSelection = this.$promisedDialog(SelectDialog, {
           itemType: "sample",
           items: this.tables.samples.items.filter(({ name }) => name)
-        }).then(selected => (selected ? [["sample", selected]] : []));
+        }).then(selected => {
+          // Fetch models related to the selected organism
+          let organismId = selected.condition.strain.organism_id;
+          let modelIds = this.modelIdsByOrganism[organismId];
+          return Promise.all(
+            modelIds.map(modelId =>
+              this.$store.dispatch("models/withFullModel", modelId)
+            )
+          ).then(() => {
+            modelIds.forEach(modelId => {
+              const model = this.getModelById(modelId);
+              const key = JSON.stringify([model.id, model.name]);
+              this.reactionsInModelsMap[key] = new Set([]);
+              model.model_serialized.reactions.forEach(reaction =>
+                this.reactionsInModelsMap[key].add(reaction.id)
+              );
+              this.metabolitesInModelsMap[key] = new Set([]);
+              model.model_serialized.metabolites.forEach(metabolite =>
+                this.metabolitesInModelsMap[key].add(
+                  getMetaboliteId(metabolite.id, metabolite.compartment)
+                )
+              );
+            });
+            return selected ? [["sample", selected]] : [];
```

Comment:
If selected can be false-y, above `selected.condition.strain.organism_id` will fail

Reply:
Add comment that we force user to choose a `sample`, so that `parsePastedMetabolites(strs)` only has to consider model ids for that sample, instead of handling different model ids because each row could have a different sample.

Reply:
`selected` cannot be falsy now when there is a specific order to upload data.
The main reason we force user to choose a `sample` is because then we can map metanetx ids to ones that are in the model, otherwise simulation will fail

Reply:
Oh, yeah, I was thinking that it was for scenario:
- already have multiple rows with differing samples (=> differing model ids for mapping)
- paste columns over those rows
- so force single sample to make mapping mnx ids easy

But actual scenario is:
- paste new rows
- if no selected sample, can't map mnx ids
- select sample later, mnx ids aren't re-mapped